rfcs/20230927-construct-tensor-without-user-handle/README.md

```diff
@@ -0,0 +1,103 @@
+# RFC: construct tensor without user handle
+
+## Introduction & Motivation
+
+In benchdnn and unit tests, users often need to allocate memory themselves. 
+Incorrect allocation methods may cause some unexpected errors, such as memory 
+release errors caused by `test::vector`. In fact, the size of the memory can be 
+obtained based on the logical tensor and layout. This RFC mainly implements the 
+ability to create tensors without user handle.
```

Comment:
```suggestion
release errors caused by `test::vector`. In fact, in scenarios like unit test, the library can take the control of allocating tensor by itself, since the size of the tensor can be 
obtained based on the logical tensor and layout. This RFC mainly implements the 
ability to create tensors without user handle.
```